test/widgets/dialog_checks.dart

```diff
@@ -46,12 +51,47 @@ Widget checkErrorDialog(WidgetTester tester, {
   }
 }
 
-/// Checks that there is no dialog.
-/// Fails if one is found.
+
+/// In a widget test, check that there aren't any alert dialogs.
+///
+/// See also:
+///  - [checkErrorDialog]
 void checkNoDialog(WidgetTester tester) {
-  check(find.byType(Dialog)).findsNothing();
-  check(find.bySubtype<AlertDialog>()).findsNothing();
-  check(find.byType(CupertinoAlertDialog)).findsNothing();
+  final List<Widget> alertDialogs = [
+    ...find.bySubtype<AlertDialog>().evaluate().map((e) => e.widget),
+    ...find.byType(CupertinoAlertDialog).evaluate().map((e) => e.widget),
```

Comment:
nit: this is equivalent, right?

```suggestion
    ...tester.widgetList(find.bySubtype<AlertDialog>()),
    ...tester.widgetList(find.byType(CupertinoAlertDialog)),
```